Relation CRUD 기능 구현

DeiLee0913 opened the pull request:
## 📝 작업 내용
> - [ ] 작업한 내용을 상세히 설명해주세요!


### 📸 결과
> - 완성한 API 결과를 첨부


## 💬 리뷰 요구사항 (선택)


## #️⃣ 연관된 이슈 (선택)
> ex) #이슈번호, #이슈번호

DeiLee0913 committed:
✨ feat: Relation 관련 Domain&Dto 생성

DeiLee0913 committed:
🌟update: JwtToken

개발을 위해 토큰 유효기간 연장

DeiLee0913 committed:
🌟update: realtion 구현 중 타 도메인 업데이트

- ProfileService와 Controller: Service의 getProfileByToken과 getProfileById의 반환값을 Long profileId에서 Profile profile로 변경(타 브랜치 변경될 예정이라 미리 반영)
- profileRepository에 Nickname으로 조회하는 쿼리 추가

DeiLee0913 committed:
✨feat: relation CRUD 기능 구현

chlqls commented on src/main/java/jusomejusome/togather/relation/service/RelationService.java:
```diff
@@ -0,0 +1,208 @@
+        }
+
+        // 현재 로그인 되어있는 user, role Id로 등록할 관계 조회
+        Role role = roleRepository.findByRoleNameAndProfile_ProfileId(reqDto.getRoleName(), profile1.getProfileId())
+                .orElseThrow(() -> new CustomException(ErrorCode.PROFILE_NOT_FOUND_EXCEPTION));
```
이 부분 예외처리 Role 관련으로 변경해야할 것 같아요~!

DeiLee0913 replied:
수정 완료했씁니다 :)

chlqls commented on src/main/java/jusomejusome/togather/relation/controller/RelationController.java:
```diff
@@ -0,0 +1,96 @@
+        return ResponseEntity.ok(allRelations);
+    }
+
+    @PutMapping("/{profileId2}")
+    public ResponseEntity<RelationResDto> updateRelationById(@AuthUser User user, @PathVariable Long profileId2, @RequestBody RelationNameReqDto nameReqDto) {
+        RelationResDto updatedRelation = relationService.updateRelationById(user, profileId2, nameReqDto);
+        return ResponseEntity.ok(updatedRelation);
+    }
+
+
+    @PutMapping("/by-name")
+    public ResponseEntity<RelationResDto> updateRelationByName(@AuthUser User user, @RequestBody RelationReqDto reqDto) {
+        RelationResDto updatedRelation = relationService.updateRelationByName(user, reqDto);
+        return ResponseEntity.ok(updatedRelation);
+    }
```
`updateRelationById`api와 `updateRelationByName`api가 별로 큰 차이가 없는 것 같아요! 
둘 다 profileId2와 새로바꿀 role의 이름을 받고 있는데 `profileId2`을  path로 받는지, RequestBody로 받는지의 차이밖에 없어보이는데 혹시 다른 차이점이 있을까요!??

DeiLee0913 replied:
RelationUpdateReqDto 추가해서 updateRelationByName은 updateReqDto로 처리하도록 변경했씁니다 ;>
updateRelationByName은 roleName과 nicknma을 입력 받아요.

chlqls commented on src/main/java/jusomejusome/togather/relation/controller/RelationController.java:
```diff
@@ -0,0 +1,96 @@
+        return ResponseEntity.ok("관계가 삭제되었습니다.");
+    }
+
+    @DeleteMapping("/by-name")
```
profile2의 닉네임을 받아서 relation을 삭제하는 api이니 통일성있게 `by-name` 대신 `by-nickname`으로 바꾸는게 더 좋을 것 같아욤!!

DeiLee0913 replied:
수정 완료했씁니다 :)

chlqls commented on src/main/java/jusomejusome/togather/relation/service/RelationService.java:
```diff
@@ -0,0 +1,208 @@
+    public void deleteRelationById(User user, Long relationId) {
+        verifyRelationAccess(user, relationId);
+
+        if(!relationRepository.existsById(relationId)) {
+            throw new CustomException(ErrorCode.RELATION_NOT_FOUND_EXCEPTION);
+        }
```
이미 `verifyRelationAccess`에서 한번 `RELATION_NOT_FOUND_EXCEPTION` 예외처리를 하고 있으니까 해당 코드는 삭제해도 될 것 같습니당

DeiLee0913 replied:
수정 완료했씁니다 :)

DeiLee0913 committed:
🌟update: realtion updateByName 업데이트

- RelationUpdateReqDto(nickname, roleName) 추가
- updateRelationByName은 updateReqDto로 처리
